drivers/gles2/rasterizer_canvas_gles2.cpp

```diff
@@ -616,24 +707,53 @@ void RasterizerCanvasGLES2::_canvas_item_render_commands(Item *p_item, Item *cur
 
 				Item::CommandRect *r = static_cast<Item::CommandRect *>(command);
 
-				glDisableVertexAttribArray(VS::ARRAY_COLOR);
-				glVertexAttrib4fv(VS::ARRAY_COLOR, r->modulate.components);
+				if (!cstate.complete)
+					glDisableVertexAttribArray(VS::ARRAY_COLOR);
+
+				const Color &col = r->modulate;
```

Comment:
Maybe rename to something more descriptive like "state_color", or "current_state_color".